docs/csharp/versioning.md

```diff
@@ -0,0 +1,128 @@
+---
+title: C# Versioning | C# Guide
+description: Understand how versioning works in C# and .NET
+keywords: .NET, .NET Core, C#
+author: tsolarin
+manager: wpickett
+ms.date: 2016/11/20
+ms.topic: article
+ms.prod: visual-studio-dev-14
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.assetid: aa8732d7-5cd0-46e1-994a-78017f20d861
+---
+
+# Versioning in C#
+
+## Authoring Libraries
+
+As a developer who has created .NET libraries for public use, you've most likely been in situations where you have
+to roll out new updates. How you go about this process matters a lot as you need to ensure that there's a seamless transition
+of existing code to the new version of your library. Here are several things to consider when creating a new release:
+
+### Semantic Versioning
+
+Semantic versioning (SemVer for short) is a naming convention applied to versions of your library to signify specific milestone events.
+Ideally, the version information you give your library should help developers determine the compatibility
+with their projects that make use of older versions of that same library.
+
+The most basic approach to SemVer is the 3 component format `MAJOR.MINOR.PATCH`, where:
+ 
+* `MAJOR` is incremented when you make incompatible API changes
+* `MINOR` is incremented when you add functionality in a backwards-compatible manner
+* `PATCH` is incremented when you make backwards-compatible bug fixes
+
+There are also ways to specify other scenarios like pre-release versions etc. Go [here](http://semver.org/) to get more information on SemVer
+and how best to utilize it when applying version information to your .NET library.
+
+### Application Configuration File
```

Comment:
I'd move this section below the Backwards Compatibility section.  I also think it needs a sample to demonstrate loading some data that changes the library's behavior.